packages/table/test/loadingOptionsTests.tsx

```diff
@@ -21,46 +43,130 @@ describe("Loading Options", () => {
+                                loadingOptions={columnLoadingOptions}
+                                renderCell={renderCell}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                            <Column
+                                loadingOptions={columnLoadingOptions}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                            <Column
+                                loadingOptions={columnLoadingOptions}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                        </Table>,
+                    );
+
+                    // only testing the first column of body cells because the second and third
+                    // columns are meant to test column related loading combinations
+                    const cells = tableHarness.element.querySelectorAll(".bp-table-cell.bp-table-cell-col-0");
+                    const columnHeaders = tableHarness.element
+                        .querySelectorAll(".bp-table-column-headers .bp-table-header");
+                    const rowHeaders = tableHarness.element.querySelectorAll(".bp-table-row-headers .bp-table-header");
+                    testLoadingOptionOverrides(
+                        columnHeaders,
+                        CellType.COLUMN_HEADER,
+                        isCellLoading,
+                        columnLoadingOptions,
+                        tableLoadingOptions,
+                    );
+                    testLoadingOptionOverrides(
+                        rowHeaders,
+                        CellType.ROW_HEADER,
+                        isCellLoading,
+                        columnLoadingOptions,
+                        tableLoadingOptions,
+                    );
+                    testLoadingOptionOverrides(
+                        cells,
+                        CellType.BODY_CELL,
+                        isCellLoading,
+                        columnLoadingOptions,
+                        tableLoadingOptions,
+                    );
+                });
+            });
+        });
     });
 });
+
+function generatePowerSet<T>(list: T[]) {
+    const base2 = (number: number) => number.toString(2);
+    const numberOfSubsets = Math.pow(2, list.length);
+    const listOfSubsets: T[][] = [];
+
+    for (let i = 1; i < numberOfSubsets; i++) {
+        const subset: T[] = [];
+        // front-pad the string and then slice from the back to ensure fixed length binary string
+        let binaryString = (Array(list.length).join("0") + base2(i)).slice(list.length * -1);
+        for (let j = 0; j < binaryString.length; j++) {
+            if (binaryString.charAt(j) === "1") {
+                subset.push(list[j]);
+            }
+        }
+        listOfSubsets.push(subset);
+    }
+
+    return listOfSubsets;
+}
+
+function testLoadingOptionOverrides(
```

Comment:
comments are always appreciated, esp in a file with so much complex code.

i suspect there are some nice refactors that could be done to clean this up, but i recognize that you're basically out of time here :(